Mathlib/Algebra/Notation/Pi.lean

```diff
@@ -13,56 +14,135 @@ This file provides only the notation for (pointwise) `0`, `1`, `+`, `*`, `•`,
 See `Mathlib.Algebra.Group.Pi.Basic` for the `Monoid` and `Group` instances.
 -/
 
-assert_not_exists Set.range Monoid DenselyOrdered
+assert_not_exists Monoid Preorder
 
 open Function
 
 universe u v₁ v₂ v₃
 
-variable {I : Type u}
+variable {ι α β γ : Type*} variable {M N O : ι → Type*} (x y : ∀ i, M i) (i : ι)
```

Comment:
These renames create an awful lot of downstream churn; are we sure we actually want them?

Reply:
I think the churn is worth it: `f` is a really terrible name for a type!

I am happy to perform the rename in a preliminary PR, though

Reply:
A preliminary PR would be great!